docs/proposals/PROPOSAL_H.md

```diff
@@ -0,0 +1,352 @@
+
+If platform.compatibilities is detected and not empty, then runtime is not required to send referrs API to query compatibility artifact.
+
+If there is no platform.compatibilities, or the content of platform.compatibilities is  empty, then runtime should send referrs API to registry to query compatibility artifact.
+
+## Pros. and Cons. Comparation
+
+### Pros. and Cons. of Compatibility Artifact
+
+It's possible to update compatibility independently without having to re-release and re-distribute image.
+
+In some vertical industries, re-distributing image means repeating release cycle to production site, it may take several weeks to several months for complex applications, very time consuming and high cost.
+
+The application using new image has to pass series tests inside vendor's lab and then customer's lab, and then roll out in production site gradually. Tests may include function, performance, reliability, security, inter-working etc.
+
+The cons. of compatibility artifact is, if it's used for image selection or scheduling decision, additional referrs API request should be sent to registry.
+
+### Pros. and Cons. of platform.compatibilities
+
+Obviously current API interaction with registry can retrieve the compatibility requirements directly, no additional referrs API request is needed.
+
+Modification to the platform.compatibilities will alter the image index content.
+
+Image shipping usually will try to ensure the integrity of the image, and prevent unauthorized tampering. Image index content update will require re-distribute the image for supply chain security.
+
+## Compatibility and NFD filed labels
+
+The proposal itself does not intend to define concrete compatibility field labels, it should be a task for separate proposals.
+
+To free the tools implementation, and make sure the existing os.version, os.features, platform.features can be easily migrated and smoothly deprecated, it's reasonable that minimal set of OCI compatibility fields should be defined in OCI.
+
+These field labels could be easily mapped or translated to NFD field labels.
+
+Because the value of compatibility field SHOULD be described in contiguous or discontinuous range if the value discovered on nodes can be different, NFD labels are dynamicly created according to node features, it's not one to one mapping.
+
+For example, oci.kernel.configurations may containes serveral boot configurations in a single string, but in NFD, different lables like kernel-config.PREEMPT, kernel-config.intel_iommu may be applied.
+
+If the field is required in image compatibilities, but does not present in NFD labels, the new feature lable should be discussed and contributed into NFD.
+
+It's expected that NFD feature lables definition and node feature discovering could be an independent component in NFD, and could be consumed by different consumers.
```

Comment:
Nit: "labels" is misspelled as "lables" twice here.

Reply:
not only "lables" spelling issue... I should use tools to check spelling and grammar next time.

Reply:
With VS Code, the "Code Spell Checker" plugin has saved me so many times.